Woof/WoofTests/ViewModel/SitterListViewModelTests/SitterListViewModelTests.swift

```diff
@@ -0,0 +1,121 @@
+import NetworkService
+import XCTest
+
+final class SitterListViewModelTests: XCTestCase {
+    override func setUp() {
+        super.setUp()
+        URLProtocol.registerClass(MockURLProtocol.self)
+        viewModel = SitterListViewModel()
+    }
+
+    override func tearDown() {
+        URLProtocol.unregisterClass(MockURLProtocol.self)
+        viewModel = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testViewModelStoresSittersWhichSentFromServer() async throws {
+        // Given
+        let data = try getData(fromJSON: "DataWithSitters")
+        MockURLProtocol.requestHandler = { request in
+            let url = try XCTUnwrap(request.url)
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: url,
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, data)
+        }
+
+        // When
+        await viewModel.fetchSitters()
+
+        // Then
+        XCTAssertEqual(Set(viewModel.sitters), DummyServerResponseData.sitters)
+    }
+
+    func testViewModelStoresInitialValueWhenResponseFromServerHasNoSittersData() async throws {
+        // Given
+        MockURLProtocol.requestHandler = { request in
+            let url = try XCTUnwrap(request.url)
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: url,
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        // When
+        await viewModel.fetchSitters()
+
+        // Then
+        XCTAssertEqual(viewModel.sitters, [])
+    }
+
+    func testViewModelStoresInitialValueWhenResponseFromServerContainsNotConvertableIntoSittersData() async throws {
+        // Given
+        MockURLProtocol.requestHandler = { request in
+            let url = try XCTUnwrap(request.url)
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: url,
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, Data())
+        }
+
+        // When
+        await viewModel.fetchSitters()
+
+        // Then
+        XCTAssertEqual(viewModel.sitters, [])
+    }
```

Comment:
We have an empty array by default at ViewModel, regarding to test name may be better to assign some value to sitters property and check that it will not change
```suggestion
        let array = Array(DummyServerResponseData.sitters)
        viewModel.sitters = array
        // Given
        MockURLProtocol.requestHandler = { request in
            let url = try XCTUnwrap(request.url)
            let response = try XCTUnwrap(
                HTTPURLResponse(
                    url: url,
                    statusCode: 200,
                    httpVersion: nil,
                    headerFields: nil
                )
            )
            return (response, nil)
        }

        // When
        await viewModel.fetchSitters()

        // Then
        XCTAssertEqual(viewModel.sitters, array)
```

Reply:
Excellent idea!
Fixed.